packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/Tasks.js

```diff
@@ -24,67 +25,74 @@ function storeMapper (stores) {
   }
 }
 
-class Tasks extends React.Component {
-  [asyncStates.initialized] () {
-    return null
-  }
-
-  [asyncStates.loading] () {
-    return (<Paragraph>Loading</Paragraph>)
-  }
+function Tasks (props) {
+  const stores = useStores()
+  console.log('stores', stores)
+  const {
+    classification,
+    isThereTaskHelp,
+    loadingState,
+    subjectReadyState,
+    step,
+    theme
+  } = storeMapper(stores)
 
-  [asyncStates.error] () {
-    console.error('There was an error loading the workflow steps and tasks.')
-    return (<Paragraph>Something went wrong</Paragraph>)
-  }
+  const stateRenderFunctions = {
```

Comment:
This object will break if the strings set in `asyncStates` change. I think you can use dynamic object properties to map states to handlers:
```js
function onError() {
…error handling
}
function onSuccess() {
…success stuff.
}
const stateRenderFunctions = {
  [asyncStates.error]: onError,
  [asyncStates.success]: onSuccess
}
```

Alternatively, maybe use dynamic method naming, like the original class did?

```js
const stateRenderFunctions = {
  [asyncStates.error] () {
   …error handling stuff
  },
  [asyncStates.success] () {
   …render the tasks
  }
}
```

Reply:
I tried that yesterday but ran into type errors. I can try again.